edb/server/protocol/auth_ext/oauth.py

```diff
@@ -30,44 +30,39 @@ class Client:
     provider: base.BaseProvider
 
     def __init__(
-        self,
-        db: Any,
-        provider_name: str,
-        base_url: str | None = None
+        self, db: Any, provider_name: str, base_url: str | None = None
     ):
         self.db = db
 
         http_factory = lambda *args, **kwargs: http_client.HttpClient(
             *args, edgedb_test_url=base_url, **kwargs
         )
 
+        provider_args = self._get_provider_config(provider_name)
+        provider_kwargs = {
+            "http_factory": http_factory,
+            "additional_scope": self._get_additional_scope(provider_name),
+        }
+
```

Comment:
Try declaring `provider_class: Type[base.BaseProvider]` here, before `match`

Reply:
Yeah, [I did that](https://github.com/edgedb/edgedb/pull/6234#discussion_r1347648942) and it still gave me the exact same error.

Reply:
Not exactly same:

```
edb/server/protocol/auth_ext/oauth.py:61:58: error: Argument 2 to "BaseProvider" has incompatible type "**Dict[str, object]"; expected "str"  [arg-type]
edb/server/protocol/auth_ext/oauth.py:61:58: error: Argument 2 to "BaseProvider" has incompatible type "**Dict[str, object]"; expected "Optional[str]"  [arg-type]
edb/server/protocol/auth_ext/oauth.py:61:58: error: Argument 2 to "BaseProvider" has incompatible type "**Dict[str, object]"; expected "Callable[..., HttpClient]"  [arg-type]
```

You can silence this one though for now as you're doing dynamic argument passing and properly typing that is only possible in 3.12

Reply:
Hmm... when I tried it locally it gave me literally the same error, maybe I was in a weird state with unsaved files or something. Fixed in [`8483e46` (#6234)](https://github.com/edgedb/edgedb/pull/6234/commits/8483e46fadc409a525d05c68ffb13e25a5768d6e)